### ISSUE TITLE
one shot CNN for both detection and feature extraction

### ISSUE THREAD

**Issue by Grabber** [nwojke/deep_sort#7]
Is there any code showing the loss function to train the current residual network?
What the authors think about having a single network to do detection and feature extraction? Is it possible?

**Comment by nwojke** [comment]
The training code is not yet published, but that should be a follow-up at some point.

About detection and feature extraction in a single network: We did some initial experiments where we fine-tuned the final layer of a pre-trained py-faster-rcnn on VOC, but it did not work that well for re-identification / tracking. It would be interesting if you get it reasonable results, though.

**Comment by Grabber** [comment]
@nwojke

Are you using a cosine loss function on the network that is extracting features for re-identification? Could you share it, please?

What I'm thinking about is use object detection patch-features as the input for re-identification network, on a single shot. The object detection features already know how to describe pedestrians, for example. I think it is a better idea and just feeding image pedestrian-patches to a new network.

What do you think?

**Comment by nwojke** [comment]
Sorry for the late reply. We do use a cosine loss function, but haven't found the time to release the training code yet. I will let you know when we get there. Until then, you could experiment with some other well established loss formulations (e.g., [1,2]) and check how well they do when providing as input the faster-rcnn features. The tracker supports changing from cosine to Euclidean metric, just change "cosine" to "euclidean" in [deep_sort_app.py:162](https://github.com/nwojke/deep_sort/blob/master/deep_sort_app.py#L162).

[1] https://arxiv.org/abs/1703.07737
[2] https://arxiv.org/abs/1511.05939

**Comment by nwojke** [comment]
Closing this one down due to inactivity.

**Comment by tonmoyborah** [comment]
@nwojke Is the code for training on objects other than pedestrians out by now?

**Comment by ovgeorge** [comment]
Is the code for training on objects other than pedestrians out by now?
